### PR TITLE
[AIRFLOW-48] Parse connection uri querystring as extra_dejson

### PR DESCRIPTION
**Pull request by inytar**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-48
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
  - This PR makes it so that the uri query string is parsed as dictionary, turned to json and placed in the `extra` property allowing for example to set the `sslmode` on a Postgres connection.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
  - Added unit tests for parsing the connection from a uri.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: tests/models.py [hunk 1]
```diff
@@ -2326,3 +2326,27 @@ def test_connection_extra_with_encryption(self, mock_get):
...
+        uri = 'scheme://user:password@host%2flocation:1234/schema?'\
+            'extra1=a%20value&extra2=%2fpath%2f'
+        connection = Connection(uri=uri)
+        self.assertEqual(connection.conn_type, 'scheme')
+        self.assertEqual(connection.host, 'host/location')
+        self.assertEqual(connection.schema, 'schema')
+        self.assertEqual(connection.login, 'user')
+        self.assertEqual(connection.password, 'password')
+        self.assertEqual(connection.port, 1234)
+        self.assertEqual(connection.extra_dejson, {'extra1': 'a value',
```
Comment (gglanzani) [inline tests/models.py]: You should use `self.assertDictEqual` here.

**Comment by inytar** [comment]
@gglanzani I've updated the test, could you look at this again?

**Comment by gglanzani** [comment]
@inytar Hi Pieter, LGTM. Could you squash your commits (`git rebase -i HEAD~2`)?

**Comment by inytar** [comment]
Of course!

**Comment by gglanzani** [comment]
Pinging @Fokko

**Review by Fokko — APPROVED** [review]
Thanks @gglanzani and @inytar

**Pull request closed without merging**